appium/sample-scripts/python/device_finder.py

```diff
@@ -65,9 +67,16 @@ def available_android_device(self, limit=0):
                     print ""
                     return str(device['displayName'])
 
-        print "No available device found"
-        print ""
-        return ""
+        json_resp = self.get_devices(limit, ";s_osType_eq_android" + free)
```

Comment:
What does the variable "free" mean in this context? I didn't find any init of this var from the whole script.

Currently it looks like this script forces any premium available device to be used, if all free ones or device group devices are unavailable. Shouldn't this be communicated somehow to end-user? Would be good to have also a comment line indicating what is going on if the for loop ends without returning a device.